src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala

```diff
@@ -110,4 +123,110 @@ class UserService(rawlsDAO: RawlsDAO, thurloeDAO: ThurloeDAO, userToken: UserInf
+        "could not save Anonymous Group", exception))
+      case Success(_) => {
+        val futureAllKeys: Future[ProfileWrapper] = getAllKeysFromThurloe(userToken)
+        futureAllKeys map { keys => RequestComplete(keys)}
+      }
+    }
+  }
+
+  private def getAnonymousGroup: String = {
+    // randomly generate the anonymousGroupName, which follows format: terra-user-adjective-noun-endOfUUID
+    val anonymousGroupUUID: UUID = UUID.randomUUID()
+    val anonymousGroupName: String = (FireCloudConfig.FireCloud.supportPrefix
+      + getWord(anonymousGroupUUID.getMostSignificantBits(), UserService.randomAdjectiveList) + "-"
+      + getWord(anonymousGroupUUID.getLeastSignificantBits(), UserService.randomNounList) + "-"
+      + anonymousGroupUUID.toString().split("-")(4))
+    anonymousGroupName
+  }
+
+  private def getAllKeysFromThurloe(userToken: UserInfo): Future[ProfileWrapper] = {
+    thurloeDAO.getAllKVPs(userToken.id, userToken) map { // .getAllKVPs returns Option[ProfileWrapper]
+      case None => ProfileWrapper(userToken.id, List())
+      case Some(wrapper) => {
+        wrapper
+      }
+    }
+  }
+
+  /**
+    * creates a new anonymized Google group for the user and adds the user's contact email to the new Google group
+    * @param keys                 the user's KVPs
+    * @param anonymousGroupName   sets the name of the Google group to be created
+    * @return                     Future[PerRequestMessage] for all KVPs for the user
+    */
+  private def setupAnonymizedGoogleGroup(keys: ProfileWrapper, anonymousGroupName: String): Future[PerRequestMessage] = {
+    // define userEmail to add to google Group - check first for contactEmail, otherwise use user's login email
+    val userEmail = getProfileValue(keys, UserService.ContactEmailKey) match {
+      case None | Some("") => userToken.userEmail
+      case Some(contactEmail) => contactEmail // if there is a non-empty value set for contactEmail, we assume contactEmail is a valid email
+    }
+
+    // create the new anonymized Google group
+    googleServicesDAO.createGoogleGroup(anonymousGroupName) match { // returns "" if group creation is not successful
+      case None | Some("") => {
+        Future(RequestComplete(keys))
+      }
+      case Some(groupEmailName) => {
+        // if Google group creation was successful, add the user's email address to the group
+        googleServicesDAO.addMemberToAnonymizedGoogleGroup(groupEmailName, userEmail) match { // returns "" if user addition is not successful
+          case None | Some("") => {
+            Future(RequestComplete(keys))
+          }
+          case Some(addedUserEmail) => {
+            // only if the anonymized Google group was successfully created and user email added to group
+            writeAnonymousGroup(userToken, groupEmailName) // write new KVP to Thurloe
+          }
+        }
+      } // this returns a Future[PerRequestMessage]
```

Comment:
I don't think you need this comment - the return of the last execution is standard, and `Future[PerRequestMessage]` is defined as the return value in the method signature.